### DISCUSSION TITLE
How many options in `select` possible?

### DISCUSSION THREAD

**Discussion by jannismilz** [pocketbase/pocketbase#1127] [Q&A] [answered]
Are there any benchmarks or experiences about the `select` field? Like is there a limit or an amount where it starts to get noticeable slower?
How many options are recommended to put in there?

**Comment by ganigeorgiev** [comment, marked as answer]
I haven't done any benchmarks for that.

In general, you can have as many options as you want (internally they are stored as serialized json array) but if you plan to have more than 20-30 `select` field options, you may be better with changing the field to a `relation` and convert the options to records. This way you will have more control on the options list and it will be easier to change (including adding additional attributes to a single option/record like title, description, etc.).

**Reply by jannismilz** [reply]
Thanks alot for this advice!

**Reply by ganigeorgiev** [reply]
_Just for info, you could also check the default SQLite limits in https://www.sqlite.org/limits.html._

**Reply by jannismilz** [reply]
@ganigeorgiev Can I filter the select fields as well with the filter parameter?

**Reply by ganigeorgiev** [reply]
You can filter a **single select** field value like any other field, eg. `mySelectField = 'active'`, but **multi-select** field are matched as it is and you can't do direct lookups for their individual values. There will be support for that in the near future, but for now if you have multi-select field you can do loose checks via the `~` operator, eg. `myMultiSelectField ~ "active"`, where an example matching `myMultiSelectField` value would be `["active", "completed"]`.

**Reply by jannismilz** [reply]
Awesome thanks
